proto/uber/cadence/api/v1/tasklist.proto

```diff
@@ -57,12 +57,24 @@ message TaskListPartitionMetadata {
   string owner_host_name = 2;
 }
 
+message IsolationGroupMetrics {
+  // The rate at which tasks with the given isolation group are being added to this partition
+  double new_tasks_per_second = 1;
+  // The number of "active" pollers with the given isolation group that the partition is aware of.
+  // Pollers typically will reach many partitions
+  int64 poller_count = 2;
+}
+
 message TaskListStatus {
   int64 backlog_count_hint = 1;
   int64 read_level = 2;
   int64 ack_level = 3;
+  // The rate at which tasks may be dispatched according to poller configuration
   double rate_per_second = 4;
   TaskIDBlock task_id_block = 5;
+  map<string, IsolationGroupMetrics> isolation_group_metrics = 6;
+  // The rate at which tasks are being added to this partition
+  double new_tasks_per_second = 7;
```

Comment:
I notice the isolation_group_metrics also has a new_tasks_per_second field, are those two field's going to have the same value or different value?
- If they are going to be the same, do we necessary needs two fields?
- If different, maybe we can rename the new_tasks_per_second in isolationGroupMetrics to be "isolation_group_new_tasks_per_second"?

Additionally, the existing rate_per_second might be confused with the new_tasks_per_second. In my understanding, the new_tasks_per_second might override the existing rate_per_second? We might need more clarification there.

Reply:
I've updated the comments to better clarify the intention. The two `new_tasks_per_second` fields represent the same value but in different contexts: In `IsolationGroupMetrics` it's the value for an individual isolation group, in `TaskListStatus` it represents the entire partition of the task list. I think them having the same name is reasonable since they're conceptually the same value. It's similar to how both a `Circle` and `Square` might have a field called `area`.

Agreed that the existing `rate_per_second` field is confusing. I've added comments to both fields to better distinguish them. They represent entirely different values. The `rate_per_second` is a rate limit on how quickly the task list can dispatch tasks. Users can set it in their worker configuration, which the poller sends to the server with each request.